mono/mini/method-to-ir.c

```diff
@@ -15053,6 +14989,72 @@ mono_spill_global_vars (MonoCompile *cfg, gboolean *need_local_opts)
 	g_free (live_range_end_bb);
 }
 
+static void mono_decompose_typecheck (MonoCompile *cfg, MonoBasicBlock *bb, MonoInst *ins)
```

Comment:
I'll do that in another PR. Most util functions used here are `static`, it will require a bit more moving code around.